docs/tutorial.rst

```diff
@@ -433,6 +433,31 @@ The *context* variable in all cases is an instance of
 
 .. _docid.tutorial.environmental-controls:
 
+Using Context data in steps
+---------------------------
+
+Like shown above, you can use the data in "context" to pass data between steps.
+Variables stored there can also be used in a scenario like so:
+
+.. code-block:: python
+
+    @given('I request a new widget for an account via SOAP')
+    def step_impl(context):
+        client = Client("http://127.0.0.1:8000/soap/")
+        response = client.Allocate(first='Firstname', last='Lastname')
+        context.widget_id = response['id']
+
+    @then('I delete the widget with ID "{id}"')
+    def step_impl(context, id):
+        client = Client("http://127.0.0.1:8000/soap/")
+        client.delete(id)
```

Comment:
I understand this example is about showcasing a typical example.

Just for clarification, let me ask a few questions:

- Why would we want to use the `widget_id` in the Gherkin-language feature file? (What if the attribute name changes?)
- Wouldn't it be wiser to use the `context.widget_id` directly in the code, instead of passing it in via the `id` parameter? (Use [declarative style](https://cucumber.io/docs/bdd/better-gherkin/#consider-a-more-declarative-style)!)

I'm not trying to make "stupid questions". Maybe I'm missing something, or the example could be improved.

What do you think?